Sources/HTTPManager.swift

```diff
@@ -1620,6 +1620,10 @@ extension HTTPManager {
             switch uploadBody {
             case .data(let data)?:
                 networkTask = inner.session.uploadTask(with: urlRequest, from: data)
+            case .formUrlEncoded(let queryItems)?:
+                networkTask = inner.session.uploadTask(with: urlRequest, from: FormURLEncoded.data(for: queryItems))
+            case .json(let json)?:
+                networkTask = inner.session.uploadTask(with: urlRequest, from: JSON.encodeAsData(json))
```

Comment:
What I meant by this is you're encoding JSON inside the `inner.sync` block, which is a critical section. If you do it at the top of the function, where `.formUrlEncoded` is handled (i.e. convert the `uploadBody` to a `.data`), then you'll get Content-Length (and the cost of encoding JSON on the main thread), it just won't be inside the critical section.